debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/YugabyteDBConnector.java

```diff
@@ -61,10 +62,81 @@ public void start(Map<String, String> props) {
 
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
+        if (props == null) {
+            return Collections.emptyList();
+        }
+
+        if (props.containsKey(PostgresConnectorConfig.SNAPSHOT_MODE.name())
+                && props.get(PostgresConnectorConfig.SNAPSHOT_MODE.name())
+                    .equalsIgnoreCase(PostgresConnectorConfig.SnapshotMode.PARALLEL.getValue())) {
+            LOGGER.info("Initialising parallel snapshot consumption");
+
+            final String tableIncludeList = props.get(PostgresConnectorConfig.TABLE_INCLUDE_LIST.name());
+            // Perform basic validations.
+            validateSingleTableProvidedForParallelSnapshot(tableIncludeList);
+
+            // Publication auto create mode should not be for all tables.
```

Comment:
Can also move this into a separate function - `validatePublicationForParallelSnapshot`